doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -144,61 +144,129 @@ presentation of Cic is given in Chapter :ref:`calculusofinductiveconstructions`.
-                    : let `ident` [`binders`] [: `term`] := `term` in `term`
-                    : let fix `fix_body` in `term`
-                    : let cofix `cofix_body` in `term`
-                    : let ( [`name` , … , `name`] ) [`dep_ret_type`] := `term` in `term`
-                    : let ' `pattern` [in `term`] := `term` [`return_type`] in `term`
-                    : if `term` [`dep_ret_type`] then `term` else `term`
-                    : `term` : `term`
-                    : `term` <: `term`
-                    : `term` :>
-                    : `term` -> `term`
-                    : `term` `arg` … `arg`
-                    : @ `qualid` [`term` … `term`]
-                    : `term` % `ident`
-                    : match `match_item` , … , `match_item` [`return_type`] with
-                    :   [[|] `equation` | … | `equation`] end
-                    : `qualid`
-                    : `sort`
-                    : `num`
-                    : _
-                    : ( `term` )
-   arg              : `term`
-                    : ( `ident` := `term` )
-   binders          : `binder` … `binder`
-   binder           : `name`
-                    : ( `name` … `name` : `term` )
-                    : ( `name` [: `term`] := `term` )
-                    : ' `pattern`
-   name             : `ident` | _
-   qualid           : `ident` | `qualid` `field`
-   sort             : SProp | Prop | Set | Type
-   fix_bodies       : `fix_body`
-                    : `fix_body` with `fix_body` with … with `fix_body` for `ident`
-   cofix_bodies     : `cofix_body`
-                    : `cofix_body` with `cofix_body` with … with `cofix_body` for `ident`
-   fix_body         : `ident` `binders` [`annotation`] [: `term`] := `term`
-   cofix_body       : `ident` [`binders`] [: `term`] := `term`
-   annotation       : { struct `ident` }
-   match_item       : `term` [as `name`] [in `qualid` [`pattern` … `pattern`]]
-   dep_ret_type     : [as `name`] `return_type`
-   return_type      : return `term`
-   equation         : `mult_pattern` | … | `mult_pattern` => `term`
-   mult_pattern     : `pattern` , … , `pattern`
-   pattern          : `qualid` `pattern` … `pattern`
-                    : @ `qualid` `pattern` … `pattern`
-                    : `pattern` as `ident`
-                    : `pattern` % `ident`
-                    : `qualid`
-                    : _
-                    : `num`
-                    : ( `pattern` | … | `pattern` )
+   term : `binder_constr`
+        : `term100`
+   term100 : `term_cast`
+           : `term10` :>
+           : `term10`
+   term10 : `term9` `appl_arg_list`
+          : @ `qualid` `instance` `term9_list_opt`
+          : @ ? `ident` `ident_list`
+          : `term9`
+   ident_list : `ident_list` `ident`
+              : `ident`
+   appl_arg_list : `appl_arg_list` `appl_arg`
+                 : `appl_arg`
+   appl_arg : ( `ident` := `term` )
+            : `term9`
+   term9 : .. `term0` ..
```

Comment:
It would seem that this is *yet another* hack:

```
Check (.. True ..). (* Special token .. is for use in the Notation command. *)
```

Reply:
Could add a comment in the grammar.

Reply:
I think this is one of the few exceptions that should be hidden entirely. You can explain why we do this by a comment in the `edit_mlg` file.

Reply:
And not document it in the notations section??  If we document it there, I think we need to keep term9 in the grammar.

Reply:
Indeed, not document it anywhere (at least not as a grammar thing).